docs/site/reference/stdin-spec.md

```diff
@@ -0,0 +1,51 @@
+---
+outline: deep
+---
+
+# Stdin Specification
+
+Formatters **MAY** also implement the Stdin Specification, which allows
+formatting "virtual files" passed via stdin.
+
+A formatter **MUST** implement the Stdin Specification if its formatting behavior
+can depend on the name of the file being formatted.
+
+## Rules
+
+In order for the formatter to comply with this spec, it **MUST** implement the
+vanilla [Formatter Specification](/reference/formatter-spec), and additionally
+satisfy the following:
+
+### 1. `--stdin-filepath` flag
+
+The formatter's CLI **MUST** be of the form:
+
+```
+<command> [options] [--stdin-filepath <path>]
+```
+
+Where:
+
+- `<command>` is the name of the formatting tool.
+- `[options]` are any number of flags and options that the formatter accepts.
+- `--stdin-filepath <path>` is an optional flag that puts the formatter in
+  "stdin mode". In stdin mode, the formatter reads file contents from stdin
+  rather than the filesystem.
+    - The formatter _MAY_ understand `--stdin-filepath=<path>` as well, but **MUST**
+      understand the space separated variant.
```

Comment:
It would make sense to outline the purpose of the `<path>` argument explicitly. Both you and I know why it's there, but maybe not the reader.

Reply:
If the path argument is not needed by the tool, it's also valid to ignore it.

Reply:
I've taken a stab at this. Please take a look.